runtime/doc/map.txt

```diff
@@ -1431,6 +1431,41 @@ Possible values are (second column is the short name used in listing):
     -addr=other		  ?	other kind of range
 
 
+Incremental preview ~
+                                                  *:command-preview*
+It's possible to allow commands to be incrementally previewed when
+'inccommand' is enabled by using the -preview command attribute. Possible use
+cases are:
+	-preview		Always preview
+	-preview={func}		Preview only if {func} returns |TRUE|
+
+The {func} part must be a function with the following signature: >
+
+	:function {func}(CmdLine)
+
+The function is only given one argument, which is the entire command line. The
+function must return a boolean value which determines whether the command
+preview should be triggered or not. If -preview is used by itself without any
+arguments, the command is always previewed whenever the command line is
+modified.
+
+Note that every time a command needs to be previewed, Nvim executes the
+command. It's possible to check if a command is being executed for
```

Comment:
Nvim will run the command, will it handle reverting side effects if the user cancels the command? If it does what is the control flow, and what sort of side effects does Nvim revert (only current buffer, all buffers, etc)?

Reply:
Neovim will not handle reverting side effects if the user changes or cancels the command, the command has to do it by itself. Thanks for reminding me to mention it in the docs.